### PR TITLE
Support for GUTI Registration Procedure

### PR DESCRIPTION
**Pull request by vthiruveedula**

### CODE REVIEW

File: gnodeb/worker/gnbcpueworker/handler.go [hunk 1]
```diff
@@ -42,10 +42,20 @@ func HandleInitialUEMessage(gnbue *gnbctx.GnbCpUe,
 	intfcMsg common.InterfaceMessage) {
 
 	msg := intfcMsg.(*common.UuMessage)
-	sendMsg, err := test.GetInitialUEMessage(gnbue.GnbUeNgapId, msg.NasPdus[0], "")
-	if err != nil {
-		gnbue.Log.Errorln("GetInitialUEMessage failed:", err)
-		return
+	var sendMsg []byte
```
Comment (thakurajayL) [inline gnodeb/worker/gnbcpueworker/handler.go:45]: Function name is initialUEMessage. So we should not do any change here. Logically caller should decide if its initial message or UL message.

File: profile/profile.go [hunk 2]
```diff
@@ -115,6 +116,16 @@ func initEventMap(profile *profctx.Profile) {
 			common.REG_ACCEPT_EVENT:      common.REG_COMPLETE_EVENT,
 			common.PROFILE_PASS_EVENT:    common.QUIT_EVENT,
 		}
+	case GUTI_REGISTER:
+		profile.Events = map[common.EventType]common.EventType{
+			common.REG_REQUEST_EVENT:      common.AUTH_REQUEST_EVENT,
+			common.AUTH_REQUEST_EVENT:     common.AUTH_RESPONSE_EVENT,
+			common.SEC_MOD_COMMAND_EVENT:  common.SEC_MOD_COMPLETE_EVENT,
+			common.REG_ACCEPT_EVENT:       common.REG_COMPLETE_EVENT,
+			common.GUTI_REG_REQUEST_EVENT: common.REG_ACCEPT_EVENT,
```
Comment (thakurajayL) [inline profile/profile.go:125]: Does not sound right, though it was useful for you to do the test ! We should take real scenario how GUTI is used. You may like to de-register and then register again with GUTI.  We need to do some changes in gNBsim architecture to handle these kind of scenarios.

File: profile/profile.go [hunk 3]
```diff
@@ -201,6 +212,11 @@ func initProcedureList(profile *profctx.Profile) {
 	switch profile.ProfileType {
 	case REGISTER:
 		profile.Procedures = []common.ProcedureType{common.REGISTRATION_PROCEDURE}
+	case GUTI_REGISTER:
+		profile.Procedures = []common.ProcedureType{
+			common.REGISTRATION_PROCEDURE,
+			common.GUTI_REGISTRATION_PROCEDURE,
```
Comment (thakurajayL) [inline profile/profile.go:218]: Logically. Register, De-register and then Register with GUTI is what we are looking for.

File: gnodeb/worker/gnbcpueworker/handler.go (hunk 1 above)
Comment (vthiruveedula) [reply gnodeb/worker/gnbcpueworker/handler.go:45]: agree. reverted the change as it is not required for this procedure

File: profile/profile.go (hunk 3 above)
Comment (vthiruveedula) [reply profile/profile.go:222]: yes, i changed profile. please check

File: profile/profile.go (hunk 2 above)
Comment (vthiruveedula) [reply profile/profile.go]: yes, i changed a profile and also changed some logic to send guti registration request after deregistration

**Comment by gab-arrobo** [comment]
@vthiruveedula can you please rebase this PR? Thanks!

**Comment by thakurajayL** [comment]
Close this pull request. End to end code is not complete. Handling of Identity req/rsp is missing.

**Pull request closed without merging**